### ISSUE TITLE
[studio] Sync db with repository is not filtering out deleted users when extracting committer username

### ISSUE THREAD

**Issue by dejan-brkic** [craftercms/craftercms#4281]
## Describe the bug
Sync db with the repository is not filtering out deleted users when extracting a committer username. Git is storing first name last name combination when committing files. Syncer is trying to find username based on information stored in the commit. It does not exclude deleted users (users with deleted flag)

```
[ERROR] 2020-09-08T16:37:12,049 [Studio_Publisher_Thread_Pool3] [job.PublisherTask] | Failed to sync database from repository for site test-site 
org.mybatis.spring.MyBatisSystemException: nested exception is org.apache.ibatis.exceptions.TooManyResultsException: Expected one result (or null) to be returned by selectOne(), but found: 2
    at org.mybatis.spring.MyBatisExceptionTranslator.translateExceptionIfPossible(MyBatisExceptionTranslator.java:92) ~[mybatis-spring-2.0.5.jar:2.0.5]
    at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:440) ~[mybatis-spring-2.0.5.jar:2.0.5]
    at com.sun.proxy.$Proxy43.selectOne(Unknown Source) ~[?:?]
    at org.mybatis.spring.SqlSessionTemplate.selectOne(SqlSessionTemplate.java:159) ~[mybatis-spring-2.0.5.jar:2.0.5]
    at org.apache.ibatis.binding.MapperMethod.execute(MapperMethod.java:87) ~[mybatis-3.5.5.jar:3.5.5]
    at org.apache.ibatis.binding.MapperProxy$PlainMethodInvoker.invoke(MapperProxy.java:152) ~[mybatis-3.5.5.jar:3.5.5]
    at org.apache.ibatis.binding.MapperProxy.invoke(MapperProxy.java:85) ~[mybatis-3.5.5.jar:3.5.5]
    at com.sun.proxy.$Proxy56.getUserByGitName(Unknown Source) ~[?:?]
    at org.craftercms.studio.impl.v2.service.security.internal.UserServiceInternalImpl.getUserByGitName(UserServiceInternalImpl.java:376) ~[classes/:3.2.0-SNAPSHOT]
    at org.craftercms.studio.impl.v1.service.site.SiteServiceImpl.syncDatabaseWithRepo(SiteServiceImpl.java:1234) ~[classes/:3.2.0-SNAPSHOT]
    at org.craftercms.studio.impl.v1.service.site.SiteServiceImpl.syncDatabaseWithRepo(SiteServiceImpl.java:1196) ~[classes/:3.2.0-SNAPSHOT]
    at jdk.internal.reflect.GeneratedMethodAccessor206.invoke(Unknown Source) ~[?:?]
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
    at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
    at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344) ~[spring-aop-5.2.6.RELEASE.jar:5.2.6.RELEASE]
    at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198) ~[spring-aop-5.2.6.RELEASE.jar:5.2.6.RELEASE]
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.2.6.RELEASE.jar:5.2.6.RELEASE]
    at org.springframework.aop.framework.adapter.MethodBeforeAdviceInterceptor.invoke(MethodBeforeAdviceInterceptor.java:56) ~[spring-aop-5.2.6.RELEASE.jar:5.2.6.RELEASE]
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:175) ~[spring-aop-5.2.6.RELEASE.jar:5.2.6.RELEASE]
    at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:95) ~[spring-aop-5.2.6.RELEASE.jar:5.2.6.RELEASE]
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.6.RELEASE.jar:5.2.6.RELEASE]
    at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:212) ~[spring-aop-5.2.6.RELEASE.jar:5.2.6.RELEASE]
    at com.sun.proxy.$Proxy52.syncDatabaseWithRepo(Unknown Source) ~[?:?]
    at org.craftercms.studio.impl.v1.service.deployment.job.PublisherTask.syncRepository(PublisherTask.java:194) ~[classes/:3.2.0-SNAPSHOT]
    at org.craftercms.studio.impl.v1.service.deployment.job.PublisherTask.run(PublisherTask.java:119) [classes/:3.2.0-SNAPSHOT]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
    at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: org.apache.ibatis.exceptions.TooManyResultsException: Expected one result (or null) to be returned by selectOne(), but found: 2
    at org.apache.ibatis.session.defaults.DefaultSqlSession.selectOne(DefaultSqlSession.java:80) ~[mybatis-3.5.5.jar:3.5.5]
    at jdk.internal.reflect.GeneratedMethodAccessor67.invoke(Unknown Source) ~[?:?]
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
    at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
    at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:426) ~[mybatis-spring-2.0.5.jar:2.0.5]
    ... 26 more
```

**Comment by dejan-brkic** [comment]
Done
PR:
https://github.com/craftercms/studio/pull/2168
https://github.com/craftercms/studio/pull/2169

**Comment by dejan-brkic** [comment]
@yacdaniel Description of the issue 

```
Hi, we found an issue while syncing site:
The situation is that in user table, we have some deleted users. 
I noticed that we may also have similar issue if we have user with same first name and last name.
```

**Comment by yacdaniel** [comment]
Verified fixed in `3.1.10-SNAPSHOT`

**Comment by yacdaniel** [comment]
Verified fixed.
